### ISSUE TITLE
Undefined name: 'vocab' in mxnet.py

### ISSUE THREAD

**Issue by cclauss** [d2l-ai/d2l-en#948]
@astonzhang https://github.com/d2l-ai/d2l-en/blob/master/d2l/mxnet.py#L2222-L2223 on these lines `vocab` is an undefined name.

[flake8](http://flake8.pycqa.org) testing of https://github.com/d2l-ai/d2l-en on Python 3.8.3

$ __flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics__
```
./d2l/mxnet.py:2222:24: F821 undefined name 'vocab'
    premise = np.array(vocab[premise], ctx=d2l.try_gpu())
                       ^
./d2l/mxnet.py:2223:27: F821 undefined name 'vocab'
    hypothesis = np.array(vocab[hypothesis], ctx=d2l.try_gpu())
                          ^
2     F821 undefined name 'vocab'
2
```
https://flake8.pycqa.org/en/latest/user/error-codes.html

On the flake8 test selection, this PR does _not_ focus on "_style violations_" (the majority of flake8 error codes that [__psf/black__](https://github.com/psf/black) can autocorrect).  Instead these tests are focus on runtime safety and correctness:
* F82 tests are almost always _undefined names_ which are usually a sign of a typo, missing imports, or code that has not been ported to Python 3.  These also would be compile-time errors in a compiled language but in Python a __NameError__ is raised which will halt/crash the script on the user.
* E9 tests are about Python syntax errors usually raised because flake8 can not build an Abstract Syntax Tree (AST).  Often these issues are a sign of unused code or code that has not been ported to Python 3.  These would be compile-time errors in a compiled language but in a dynamic language like Python they result in the script halting/crashing on the user.
* F63 tests are usually about the confusion between identity and equality in Python.  Use ==/!= to compare str, bytes, and int literals is the classic case.  These are areas where __a == b__ is True but __a is b__ is False (or vice versa).  Python >= 3.8 will raise SyntaxWarnings on these instances.
* F7 tests logic errors and syntax errors in type hints

**Comment by goldmermaid** [comment]
Hi thanks for raising the issue. I had a PR https://github.com/d2l-ai/d2l-en/pull/951.

**Comment by xiaotinghe** [comment]
PR #953

**Comment by astonzhang** [comment]
Thanks @goldmermaid and @xiaotinghe . It's fixed now.

**Comment by cclauss** [comment]
Fixed in #953
